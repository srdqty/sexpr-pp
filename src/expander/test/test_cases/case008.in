(let-syntax
(
  (or 
    (lambda (stx)
      (letrec
      (
        (first-arg (car (cdr (stx-e stx))))
        (second-arg (car (cdr (cdr (stx-e stx)))))
        (binding (mk-stx
                    (list (syntax it) first-arg)
                    stx))
      )
        (mk-stx
          (list (syntax letrec)
                (mk-stx (list binding) stx)
                (mk-stx
                  (list (syntax if)
                        (syntax it)
                        (syntax it)
                        second-arg)
                  stx))
          stx)
      )
    ))
)
  (letrec
    ((it #f))
    (or it 3)))
