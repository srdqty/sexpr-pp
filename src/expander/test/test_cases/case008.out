>> (begin (letrec ((it0 #f)) (letrec ((it1 it0)) (if it1 it1 3))))
>> 
